src/bridge/cockpitcpusamples.c

```diff
@@ -132,48 +132,65 @@ read_file (const gchar *path)
 }
 
 static void
-sample_cpu_sensors (CockpitSamples *samples,
-                    int hwmonID)
+sample_cpu_sensors (gchar *sensor_path,
+                    CockpitSamples *samples)
+{
+    g_autofree gchar *temp_content = read_file (sensor_path);
+
+    if (temp_content == NULL)
+        return;
+
+    gint64 temperature = g_ascii_strtoll (temp_content, NULL, 10);
+    // overflow or invalid
+    if (temperature == G_MAXINT64 || temperature == G_MININT64 || temperature == 0)
+      {
+        g_debug("Invalid number in %s: %m", sensor_path);
+        return;
+      }
+
+    cockpit_samples_sample (samples, "cpu.temperature", sensor_path, temperature/1000);
+}
+
+static void
+add_cpu_sensor_files (GList **devices,
```

Comment:
Perhaps detect_cpu_sensor() for more symmetry, or detect_one_cpu_sensor() if you want to make it a bit more distinct? "_files" sounds a bit weird.

Reply:
This doesn't really detect only one sensor. `detect_cpu_sensors` finds correct hwmon and `add_cpu_sensor_files` finds all CPU related `tempY_input` files in said hwmonX. I'll think of a better naming for both of these functions.

Reply:
The new names are nicer, thank you!